ipn/store.go

```diff
@@ -14,14 +14,66 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"strings"
 	"sync"
 	"time"
 
 	"tailscale.com/atomicfile"
 	"tailscale.com/kube"
+	_ "tailscale.com/kube"
 	"tailscale.com/paths"
 )
 
+// NewStoreFunc returns a StateStore for the provided path.
+type NewStoreFunc func(path string) (StateStore, error)
+
+var knownStores = map[string]NewStoreFunc{
+	"kube:": func(path string) (StateStore, error) {
+		secretName := strings.TrimPrefix(path, "kube:")
+		return NewKubeStore(secretName)
+	},
+	"mem:": func(string) (StateStore, error) {
+		return new(MemoryStore), nil
+	},
+	"file:": func(path string) (StateStore, error) {
+		path = strings.TrimPrefix(path, "file:")
+		return NewFileStore(path)
+	},
+}
+
+// ResolveStore returns a StateStore based on the provided path
+// and registered stores. It returns nil when there is no matching
```

Comment:
what part is nil? (nil, nil)? Why is that a good result? That seems bogus in like all cases.

Reply:
Returning an error now.